bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/progress/UIJob.java

```diff
@@ -31,6 +34,53 @@
  * @since 3.0
  */
 public abstract class UIJob extends Job {
+
+	/**
+	 * Creates a new UIJob that will execute the provided function in the UI thread
+	 * when it runs.
+	 *
+	 * Prefer using {@link UIJob#createUI(String, ICoreRunnable)}.
```

Comment:
The Job class recommends to prefer the ICoreRunnable overload, but the mentioned reason is not valid anymore because both get monitor.done() called and can use `OperationCanceledException`.

Therefore I wonder if the recommendation is still required or if just the `ICoreRunnable` should be added?

Reply:
Since we have a factory with `IJobFunction` and `ICoreRunnable`, the method calls is ambiguous for non-block lambda expressions, which is a bit ugly because then one has to cast the expression to `IJobFunction` or `ICoreRunnable`.
If I only add a `ICoreRunnable` that problem wouldn't occur.
Correct me if I'm wrong, but I have the impression that `ICoreRunnable` is the more modern approach and the `IJobFunction`  factory is only in the Job class for backwards compatibility and one can achieve the same regarding error-propagation and cancellation with an ICoreRunnable. Therefore I'm tending more towards just adding the ICoreRunnable factory.